yarn/src/main/scala/org/apache/spark/deploy/yarn/ApplicationMasterArguments.scala

```diff
@@ -63,10 +63,6 @@ class ApplicationMasterArguments(val args: Array[String]) {
           userArgsBuffer += value
           args = tail
 
-        case ("--num-workers" | "--num-executors") :: IntParam(value) :: tail =>
-          numExecutors = value
```

Comment:
BTW you should also remove the `numExecutors` field from this class; that should uncover a couple of other spots where you might need to also fix things.